src/platform/packages/private/kbn-esql-editor/src/esql_editor.tsx

```diff
@@ -1065,4 +1069,8 @@ export const ESQLEditor = memo(function ESQLEditor({
   );
 
   return editorPanel;
-});
+};
+
+export const ESQLEditor = withRestorableState(ESQLEditorInternal);
+
+export type ESQLEditorProps = ComponentProps<typeof ESQLEditor>;
```

Comment:
Why this change?

Reply:
@stratoula 

Not exactly, consumers use these props. Since UnifiedSearch is like a proxy between Discover and ES|QL editor in this case, I was using these types in UnifiedSearch as:

```
esqlEditorInitialState?: ESQLEditorProps['initialState'];
onEsqlEditorInitialStateChange?: ESQLEditorProps['onInitialStateChange'];
```
If you don't agree with changes to types, I can revert it and update to:

```
esqlEditorInitialState?: ESQLEditorRestorableState;
onEsqlEditorInitialStateChange?: (nextState: ESQLEditorRestorableState) => void;
```

Reply:
Got it! Yeah I prefer what you are proposing tbh ++

Reply:
@stratoula 

Just realized that if I follow the second option then the type checks fail because the original type is used when lazy importing the editor code in `src/platform/plugins/shared/esql/public/create_editor.tsx` and we need to set these new props in `src/platform/plugins/shared/unified_search/public/query_string_input/query_bar_top_row.tsx`. 
So I'm going to keep the current implementation if it's okay with you. The types are aligned with what the component offers.